server/src/main/java/org/spine3/server/entity/Repository.java

```diff
@@ -88,7 +90,33 @@ protected BoundedContext getBoundedContext() {
     /** Returns the class of IDs used by this repository. */
     @CheckReturnValue
     protected Class<I> getIdClass() {
-        return Classes.getGenericParameterType(getClass(), ID_CLASS_GENERIC_INDEX);
+        final Class<I> idClass = Classes.getGenericParameterType(getClass(),
+                                                                 ID_CLASS_GENERIC_INDEX);
+        checkIdClass(idClass);
+
+        return idClass;
+    }
+
+    /**
+     *  Check that this class of identifiers is supported by the framework.
+     *
+     * <p>The type of entity identifiers ({@code <I>}) cannot be bound because
+     * it can be {@code Long}, {@code String}, {@code Integer}, and class implementing
+     * {@code Message}.
+     *
+     * <p>We perform the check to to detect possible programming error
+     * in declarations of entity and repository classes <em>Integer</em> we have
```

Comment:
I don't get what `<em>Integer</em>` is doing in the middle of this sentence. Perhaps, the sentence is missing something.